src/BizHawk.Client.Common/ArgParser.cs

```diff
@@ -1,221 +1,235 @@
+			OptionLoadSavestateFilePath.Description = "path; savestate which should be loaded on launch; this takes precedence over `--{OptionLoadQuicksaveSlot.Name}`";
+			OptionLuaFilePath.Description = $"implies `--{OptionOpenLuaConsole.Name}`";
+			OptionSocketServerIP.Description = $"string; IP address for Unix socket IPC (Lua `comm.socket*`); must be paired with `--{OptionSocketServerPort.Name}`";
+			OptionSocketServerPort.Description = $"int; port for Unix socket IPC (Lua `comm.socket*`); must be paired with `--{OptionSocketServerIP.Name}`";
+			OptionSocketServerUseUDP.Description = $"pass to use UDP instead of TCP for Unix socket IPC (Lua `comm.socket*`); ignored unless `--{OptionSocketServerIP.Name} --{OptionSocketServerPort.Name}` also passed";
+
+			RootCommand root = new();
+			root.Add(ArgumentRomFilePath);
+			// `--help` uses this order, so keep alphabetised by flag
+			root.Add(/* --audiosync */ OptionAVDumpAudioSync);
+			root.Add(/* --chromeless */ OptionLaunchChromeless);
+			root.Add(/* --config */ OptionConfigFilePath);
+			root.Add(/* --dump-close */ OptionAVDumpQuitWhenDone);
+			root.Add(/* --dump-frames */ OptionAVDumpFrameList);
+			root.Add(/* --dump-length */ OptionAVDumpEndAtFrame);
+			root.Add(/* --dump-name */ OptionAVDumpName);
+			root.Add(/* --dump-type */ OptionAVDumpType);
+			root.Add(/* --fullscreen */ OptionLaunchFullscreen);
+			root.Add(/* --load-slot */ OptionLoadQuicksaveSlot);
+			root.Add(/* --load-state */ OptionLoadSavestateFilePath);
+			root.Add(/* --lua */ OptionLuaFilePath);
+			root.Add(/* --luaconsole */ OptionOpenLuaConsole);
+			root.Add(/* --mmf */ OptionMMFPath);
+			root.Add(/* --movie */ OptionMovieFilePath);
+			root.Add(/* --open-ext-tool-dll */ OptionOpenExternalTool);
+			root.Add(/* --socket-ip */ OptionSocketServerIP);
+			root.Add(/* --socket-port */ OptionSocketServerPort);
+			root.Add(/* --socket-udp */ OptionSocketServerUseUDP);
+			root.Add(/* --url-get */ OptionHTTPClientURIGET);
+			root.Add(/* --url-post */ OptionHTTPClientURIPOST);
+			root.Add(/* --userdata */ OptionUserdataUnparsedPairs);
+			root.Add(/* --version */ OptionQueryAppVersion);
+
+			Parser = new CommandLineBuilder(root)
+//				.UseVersionOption() // "cannot be combined with other arguments" which is fair enough but `--config` is crucial on NixOS
+				.UseHelp()
+//				.UseEnvironmentVariableDirective() // useless
+				.UseParseDirective()
+				.UseSuggestDirective()
+//				.RegisterWithDotnetSuggest() // intended for dotnet tools
+//				.UseTypoCorrections() // we're only using the parser, and I guess this only works with the full buy-in
+//				.UseParseErrorReporting() // we're only using the parser, and I guess this only works with the full buy-in
+//				.UseExceptionHandler() // we're only using the parser, so nothing should be throwing
+//				.CancelOnProcessTermination() // we're only using the parser, so there's not really anything to cancel
+				.Build();
```

Comment:
Not sure if it's necessary to comment all options that you're not setting, but whatever. What do the `Use*Directive` calls actually accomplish though? From the very useful documentation I found they supposedly enable certain commandline syntax, but also it seems to require `EnableDirectives` and even when setting that, I couldn't get the syntax to work, so... ?

Also, what's with the version option?

Reply:
To explain, this list is [the expansion of `UseDefaults`](https://learn.microsoft.com/en-us/dotnet/api/system.commandline.builder.commandlinebuilderextensions.usedefaults?view=system-commandline#remarks). I commented the ones I didn't use so that, if and when the list changes, there'll be a record of what didn't work in the past.
I'll be honest, I didn't play with the directives. I left them in just to follow the .NET convention and in the hope that someone would find a use for them.
We already have a `--version` which works well enough. When I (deleted that and) enabled `UseVersionOption`, it failed because it saw `--config=.../config.json --version` due to the launch script I use.